### PR TITLE
[INLONG-4137][Manager] Add complex type (array, map, struct) for Hive sink

### LINKED ISSUE #4137: [Feature][Manager] Add complex type (array, map, struct) for Hive sink
### Description

Add complex type (array, map, struct) for Hive sink

### Use case

_No response_

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by LvJiancheng**
### Title Name: [INLONG-4137][Manager] Add complex type (array, map, struct) for Hive sink

where *XYZ* should be replaced by the actual issue number.

Fixes #4137 

### Motivation

*Explain here the context, and why you're making that change. What is the problem you're trying to solve?*

### Modifications

*Describe the modifications you've done.*

### Verifying this change

*(Please pick either of the following options)*

- [ ] This change is a trivial rework/code cleanup without any test coverage.

- [ ] This change is already covered by existing tests, such as:
  *(please describe tests)*

- [ ] This change added tests and can be verified as follows:

  *(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### Documentation

  - Does this pull request introduce a new feature? (yes / no)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)
  - If a feature is not applicable for documentation, explain why?
  - If a feature is not documented yet in this PR, please create a follow-up issue for adding the documentation

### CODE REVIEW

File: inlong-manager/manager-client/src/main/java/org/apache/inlong/manager/client/api/util/InlongStreamSinkTransfer.java [hunk 1]
```diff
@@ -150,15 +151,23 @@ private static StreamSink parseClickHouseSink(ClickHouseSinkResponse sinkRespons
...
+            sinkField.setFieldComment(sinkFieldResponse.getFieldComment());
+            sinkField.setFieldValue(null);
+            sinkField.setSourceFieldName(sinkFieldResponse.getSourceFieldName());
+            sinkField.setSourceFieldType(StringUtils.isBlank(sinkFieldResponse.getSourceFieldType()) ? null :
+                    FieldType.forName(sinkFieldResponse.getSourceFieldType()));
+            sinkField.setIsMetaField(sinkFieldResponse.getIsMetaField());
+            sinkField.setFieldFormat(sinkFieldResponse.getFieldFormat());
+            sinkFields.add(sinkField);
+        }
+        return sinkFields;
```
Comment (healchow) [inline inlong-manager/manager-client/src/main/java/org/apache/inlong/manager/client/api/util/InlongStreamSinkTransfer.java:171]: Remove the unused blank line at the end of the method, thanks.

File: inlong-manager/manager-client/src/main/java/org/apache/inlong/manager/client/api/util/InlongStreamSinkTransfer.java [hunk 2]
```diff
@@ -150,15 +151,23 @@ private static StreamSink parseClickHouseSink(ClickHouseSinkResponse sinkRespons
     }
 
     private static List<SinkField> convertToSinkFields(List<SinkFieldResponse> sinkFieldResponses) {
```
Comment (healchow) [inline inlong-manager/manager-client/src/main/java/org/apache/inlong/manager/client/api/util/InlongStreamSinkTransfer.java:153]: Please add Java doc for the static method, thanks.

File: inlong-manager/manager-client/src/main/java/org/apache/inlong/manager/client/api/util/InlongStreamSinkTransfer.java [hunk 3]
```diff
@@ -299,4 +312,12 @@ private static HiveSink parseHiveSink(HiveSinkResponse sinkResponse, StreamSink
         return hiveSink;
     }
 
+    public static Pair<FieldType, Object> transferfromstring(String type) {
```
Comment (healchow) [inline inlong-manager/manager-client/src/main/java/org/apache/inlong/manager/client/api/util/InlongStreamSinkTransfer.java:315]: Java doc, please.

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/sort/util/FieldInfoUtils.java [hunk 4]
```diff
@@ -95,8 +105,15 @@ public static FieldInfo parseStreamFieldInfo(InlongStreamFieldInfo streamField,
 
     public static FieldInfo parseStreamField(StreamField streamField) {
```
Comment (healchow) [inline inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/sort/util/FieldInfoUtils.java:106]: Java doc, please.

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/sort/util/FieldInfoUtils.java [hunk 5]
```diff
@@ -205,6 +222,24 @@ public static FormatInfo convertFieldFormat(String type) {
      * @return Sort field format instance
      */
     public static FormatInfo convertFieldFormat(String type, String format) {
```
Comment (healchow) [inline inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/sort/util/FieldInfoUtils.java:224]: Java doc, please.

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/sort/util/FieldInfoUtils.java [hunk 6]
```diff
@@ -205,6 +222,24 @@ public static FormatInfo convertFieldFormat(String type) {
...
+            return transferComplexType(complexType);
+        } else {
+            return transferSimpleType(baseType, format);
+        }
+    }
+
+    /**
+     * Get the FieldFormat of Sort according to simple type string
+     *
+     * @param type
```
Comment (healchow) [inline inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/sort/util/FieldInfoUtils.java:238]: If there are no comments for params, please remove them.

**Review by healchow — CHANGES_REQUESTED** [review]

File: inlong-manager/manager-client/src/main/java/org/apache/inlong/manager/client/api/util/InlongStreamSinkTransfer.java [hunk 7]
```diff
@@ -320,4 +312,11 @@ private static HiveSink parseHiveSink(HiveSinkResponse sinkResponse, StreamSink
         return hiveSink;
     }
 
+    public static Pair<FieldType, Object> trandferfromstring(String type) {
```
Comment (gong) [inline inlong-manager/manager-client/src/main/java/org/apache/inlong/manager/client/api/util/InlongStreamSinkTransfer.java]: `trandferfromstring` use CamelCase

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/sort/util/FieldInfoUtils.java [hunk 8]
```diff
@@ -263,10 +298,127 @@ public static FormatInfo convertFieldFormat(String type, String format) {
...
+    }
 
+    /**
+     * Get the FieldFormat of Sort according to complex type string
+     *
+     * @param map
+     * @return
+     */
+    private static FormatInfo transferComplexType(Map<String, String> map) {
+        FormatInfo formatInfo = null;
```
Comment (gong) [inline inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/sort/util/FieldInfoUtils.java:311]: Add a UT for `transferComplexType` and add some case data for input param Map.

**Comment by LvJiancheng** [comment]
This issue has been resolved in other PRs

**Comment by healchow** [comment]
Fixed by #5334.

**Pull request closed without merging**